### PR TITLE
Update theme.tsx

### PR DESCRIPTION
**Pull request by CodingRule**
created a H component to handle the different heading updated the highlightAll and startLegacyApp calls in Content to use the useEffect hook updated the ThemeProps interface to include a more specific type for the frontMatter property in pageOpts

should improve the codes readability maintainability and safety

### CODE REVIEW

**Comment by CodingRule** [comment]
I'm not the best But I think this could be good? if it's not atleast I tried Lol!

File: theme.tsx [hunk 1]
```diff
@@ -16,24 +16,36 @@ type LayoutProps = React.PropsWithChildren<{
   pageOpts: NextraThemeLayoutProps['pageOpts'];
 }>;
 
+type HProps = React.PropsWithChildren<{
```
Comment (manishprivet) [inline theme.tsx:19]: What is the purpose of this element? What problem does it solve?

File: theme.tsx [hunk 2]
```diff
@@ -16,24 +16,36 @@ type LayoutProps = React.PropsWithChildren<{
...
+  h6: (props) => <H level={6} {...props} />,
   blockquote: ({ children }) => <div className="highlight-box">{children}</div>,
 };
 
 const Content = ({ children }: LayoutProps) => {
   const { asPath } = useRouter();
 
-  // Re-highlights the pages on route change
-  useEffect(() => highlightJs.highlightAll(), [asPath]);
+  useEffect(() => {
```
Comment (manishprivet) [inline theme.tsx:42]: > updated the highlightAll and startLegacyApp calls in Content to use the useEffect hook

This seems like you just re-formatted it

File: theme.tsx [hunk 3]
```diff
@@ -42,7 +54,6 @@ const Content = ({ children }: LayoutProps) => {
   );
 };
 
-// @TODO: Nextra should provide better customization to FrontMatter Props
```
Comment (manishprivet) [inline theme.tsx:45]: Why are we removing this here?

**Review by ovflowd — CHANGES_REQUESTED** [review]
These changes sadly bring no value. Also not sure what the H component is doing and why it is different.

This is very likely not going to land. Closing this PR.

**Pull request closed without merging**